providers/implementations/signature/rsa.c

```diff
@@ -917,15 +915,18 @@ static int rsa_get_ctx_params(void *vprsactx, OSSL_PARAM *params)
         } else if (p->data_type == OSSL_PARAM_UTF8_STRING) {
             switch (prsactx->saltlen) {
             case RSA_PSS_SALTLEN_DIGEST:
-                if (!OSSL_PARAM_set_utf8_string(p, "digest"))
+                if (!OSSL_PARAM_set_utf8_string(p,
+                        OSSL_PKEY_RSA_PSS_SALT_LEN_DIGEST))
                     return 0;
                 break;
             case RSA_PSS_SALTLEN_MAX:
-                if (!OSSL_PARAM_set_utf8_string(p, "max"))
+                if (!OSSL_PARAM_set_utf8_string(p,
+                        OSSL_PKEY_RSA_PSS_SALT_LEN_MAX))
                     return 0;
                 break;
             case RSA_PSS_SALTLEN_AUTO:
-                if (!OSSL_PARAM_set_utf8_string(p, "auto"))
+                if (!OSSL_PARAM_set_utf8_string(p,
+                        OSSL_PKEY_RSA_PSS_SALT_LEN_AUTO))
```

Comment:
While I personally have nothing against this break of indentation style, I wonder what our current code style guide has to say about this, not to mention `util/check-format.pl`...  one way to get around that problem could be to refactor the code to this:

``` C
        } else if (p->data_type == OSSL_PARAM_UTF8_STRING) {
            const *value = NULL;

            switch (prsactx->saltlen) {
            case RSA_PSS_SALTLEN_DIGEST:
                value = OSSL_PKEY_RSA_PSS_SALT_LEN_DIGEST;
                break;
            case RSA_PSS_SALTLEN_MAX:
                value = OSSL_PKEY_RSA_PSS_SALT_LEN_MAX;
                break;
            case RSA_PSS_SALTLEN_AUTO:
                value = OSSL_PKEY_RSA_PSS_SALT_LEN_AUTO;
                break;
            default:
                if (BIO_snprintf(p->data, p->data_size, "%d", prsactx->saltlen)
                    <= 0)
                    return 0;
                break;
            }
            if (value != NULL
                && !OSSL_PARAM_set_utf8_string(p, value))
                return 0;
```

Side note: shouldn't the default case set `p->return_size`?

Reply:
err.. this is setting utf8 strings - so I dont know what you mean :)

Reply:
I have updated to use value... Thanks.

Reply:
> err.. this is setting utf8 strings - so I dont know what you mean :)

`OSSL_PARAM_set_utf8_string()` sets `return_size`.  The default case, however, doesn't use that function at all, it goes directly at the raw data, but should still set `return_size`.  So, something like this?

``` C
            default:
                {
                    int len = BIO_snprintf(p->data, p->data_size, "%d", prsactx->saltlen);

                    if (len <= 0)
                        return 0;
                    p->return_size = len;
                }
                break;
            }
```